### PR TITLE
Improvements for App Launcher

### LINKED ISSUE #409: Show full application name in tooltip
When the name of the application is too long it is cut off and you cannot differentiate applications with names that start the same

### PR DESCRIPTION
**Pull request by lfom**
- Added setting to hide the clock
- Scroll
- Size does not change when filtering (no collapse)
- Tooltips if the app name is too long (fix #409)
- Fixed search entry styling
- Go back to previous window if ESC is pressed after resetting entry field

### CODE REVIEW

**Review by PapyElGringo — COMMENTED** [review]
I'll merge it on Dev to tweak a bit, I will externalize the tooltip feature to an TooltipManager to be able to use it in other part of the shell.
Why Dummy button though ?  It's not performance free to have actor in the stage what the point of having them?
Initially I wanted to implement https://developer.gnome.org/st/stable/StScrollable.html for the scrollable feature and also handle the mouse interaction for scrolling.
But we can start with this

**Comment by lfom** [comment]
> I'll merge it on Dev to tweak a bit, I will externalize the tooltip feature to an TooltipManager to be able to use it in other part of the shell.
> Why Dummy button though ? It's not performance free to have actor in the stage what the point of having them?
> Initially I wanted to implement https://developer.gnome.org/st/stable/StScrollable.html for the scrollable feature and also handle the mouse interaction for scrolling.

Yeah, I have just tried to make the tooltip code as generic as possible so it could be a separated class later, as I wasn't sure it would be merged and the way it was it is simpler to debug.
I don't like to mess with anything related to layout as I have no control over it, and it was a pain to change the layout of TreeView in the preferences so I just went for a simpler custom scroll function for what existed.

File: src/layout/msWorkspace/msWorkspace.js [hunk 1]
```diff
@@ -246,14 +247,34 @@ var MsWorkspace = class MsWorkspace {
...
+            !this.tileableList ||
+            this.tileableList.length < 2 ||
+            this.focusTileable === this.appLauncher
+        ) {
+            return;
+        }
+        this.focusTileable(this.appLauncher);
+    }
+
+    focusPrevious() {
```
[the remark this answers is not in the document]
Comment (lfom) [reply src/layout/msWorkspace/msWorkspace.js]: Sure, whatever makes the code clearer/easier to read is a better option.

File: src/styles/stylesheet.scss [hunk 2]
```diff
@@ -15,6 +15,10 @@ $panel-opacity: 0.876; // Don't change this value since it's dynamically changed
     @if ($bgColor== 'dark') {
         @return rgba(255, 255, 255, $opacity);
     }
+
+    @if ($bgColor== 'primary') {
+        @return rgba(255, 255, 255, $opacity);
+    }
```
[the remark this answers is not in the document]
Comment (lfom) [reply src/styles/stylesheet.scss:21]: I knew. Again, as I have no control (nor want to) over the layout, I was a temporary fix that I knew would be improved/changed once merged.

**Comment by PapyElGringo** [comment]
What about the dummy button what the use for them? I feel we can remove them

**Comment by lfom** [comment]
> * Size does not change when filtering (no collapse)

It was explained above. I don't like how the container changes size when searching for an app. Actually, IMO anything that keeps moving in the UI is distracting and should be avoided. As the size of the launcher isn't constant, I used the dummy buttons to keep the launcher size constant.

**Comment by PapyElGringo** [comment]
This size changed was already fixed when you changed the verticalOffset using the `maximumRow` instead of the `NeededRow` so I still don't understand the need of the dummy buttons. I removed them and I don't see any changes in behavior. Can you check my last changes and confirm it's okay for you.

**Comment by lfom** [comment]
Oh, I see... I am still getting used to this GTK stuff, dealing with GUI is much more complicated than on Android. I was not sure how it would affect things on screen. I have trying to fix taskActiveIndicator bugs, and I have learned a lot after coding the scroll.

Looks like it is working fine, the only problem is the pop-up menu background I mentioned elsewhere.

**Pull request merged**